src/cls/CMakeLists.txt

```diff
@@ -40,8 +34,6 @@ if (WITH_RBD)
   set(cls_rbd_srcs rbd/cls_rbd.cc rbd/cls_rbd_types.cc)
   add_library(cls_rbd SHARED ${cls_rbd_srcs})
   set_target_properties(cls_rbd PROPERTIES
-    VERSION "1.0.0"
```

Comment:
I thought about this, and after inspecting the ClassHandler I concluded that versioning is not supported. 

To support versioning, code will need to be added to the way dynamic classes register, or the way classes are (re)loaded. The way it's written today, `ClassHandler::open_all_classes` will match any file starting with CLS_PREFIX and having CLS_SUFFIX, and then proceed to initializing all matched files unconditionally.

Reply:
I didn't mean any meaningful support for versioning on the OSD side -- just that it's currently possible to feed the OSD a versioned `.so` on Linux but not a versioned `.dylib` on MacOS.

Reply:
OK. I agree that an alternative solution could be to edit the code so that it would be prepared to accept the versioned .dylib on a mac just as it does on Linux.

Do you think that this is a preferred approach? I felt like mangling with the code to strip the version number would create a false impression that versions were supported

Reply:
> OK. I agree that an alternative solution could be to edit the code so that it would be prepared to accept the versioned .dylib on a mac just as it does on Linux.

I'm a bit confused.

I don't think `ClassHandler::open_all_classes()` _actually_ accepts a versioned `.so` on Linux.  On Linux, between
```
libcls_journal.so -> libcls_journal.so.1
libcls_journal.so.1 -> libcls_journal.so.1.0.0
libcls_journal.so.1.0.0
```
it would skip `libcls_journal.so.1.0.0` and `libcls_journal.so.1` symlink and load `libcls_journal.so`.  A versioned `.so` is accepted indirectly, through a non-versioned symlink.

Now, if I understand the issue correctly, on MacOS the listing changes to something like
```
libcls_journal.dylib -> libcls_journal.1.dylib
libcls_journal.1.dylib -> libcls_journal.1.0.0.dylib
libcls_journal.1.0.0.dylib
```
and `ClassHandler::open_all_classes()`, because it checks only whether the name starts with `libcls_` and ends with `.dylib`, doesn't skip anything and attempts to load all three.

> I felt like mangling with the code to strip the version number would create a false impression that versions were supported

... so the code would need to be changed to skip anything with a version, not strip the version -- and that shouldn't create any false impressions.

Reply:
OK, I stand corrected. You are absolutely right, the versioned files where version is appended on the right of the CLS_SUFFIX won't be matched by `open_all_classes`.

With that in mind, should I invest in code that would try to strip the version from the dylib name and have this logic unconditionally enabled in `open_all_classes`? Should it be conditionally compiled for Darwin? I think that some research should be made, it could be a solved case already.

Or should we stick to the approach here by removing the version id from the built libraries since we don't have any validation or functionality dependent on the versioned libraries?

Reply:
This is all up to @rzarzynski.

Personally, dropping the version on those libraries on Linux feels rather workaround-ish, even if the versioning is dummy.  If you are going to proceed with that I would suggest adding a comment in `src/cls/CMakeLists.txt` as it wouldn't be obvious to anyone opening that file in the future that lack of versioning is intended.

Reply:
I've done a small research, couldn't find any reasonable API or algorithm to strip version information from the library name. It's rather arbitrary. If I had to do this, I'd prefer to interact with the library and ask for its name via an API.